src/chigame/games/views.py

```diff
@@ -17,11 +19,38 @@ def lobby_list(request):
     context = {"object_list": lobbies}
     return render(request, "games/lobby_list.html", context)
 
+@login_required
+def lobby_join(request, pk):
+    lobby = get_object_or_404(Lobby, pk=pk)
+    joined = Lobby.objects.filter(members=request.user.id)
+    print(joined, lobby)
+    if lobby in joined:
+        return HttpResponseNotFound("Already joined.")
+    lobby.members.add(request.user)
+    return redirect(reverse("lobby-details", kwargs={"pk": lobby.id}))
+
+
+@login_required
+def leave_lobby(request, pk):
+    lobby = get_object_or_404(Lobby, pk=pk)
+    joined = Lobby.objects.filter(members=request.user.id)
+    print(joined, lobby)
+    if lobby not in joined:
+        return HttpResponseNotFound("Haven't Joined.")
+    lobby.members.remove(request.user)
+    return redirect(reverse("lobby-details", kwargs={"pk": lobby.id}))
+
 
 class ViewLobbyDetails(DetailView):
     model = Lobby
     template_name = "games/lobby_details.html"
     context_object_name = "lobby_detail"
+    """
+    def get_context_data(self, **kwargs):
+        # Call the base implementation first to get a context
+        context = super(ViewLobbyDetails, self).get_context_data(**kwargs)
+        context['user'] = self.request.user
+    """
```

Comment:
When I try to run the code on this branch (`python3 manage.py runserver`), I get this error: `django.db.migrations.exceptions.NodeNotFoundError: Migration games.0004_initial dependencies reference nonexistent parent node ('games', '0003_category_mechanic_game_bgg_id_game_complexity_and_more')`. It seems that there are currently some issues with the migrations in this branch. Once this issue is resolved, I will re-run the code to test it -- on paper, everything looks good to me except for the stylistic issues I commented on above.

Reply:
thanks! yea i honestly am fighting demons with these migration issues rn but they will be resolved soon. Thanks for your feedback!

Reply:
I can see that!! the migration issues on my branch absolutely slayed me too. I just made this review as it is now to share the style comments. good luck!!